sql/core/src/test/scala/org/apache/spark/sql/errors/QueryExecutionErrorsSuite.scala

```diff
@@ -899,6 +899,29 @@ class QueryExecutionErrorsSuite
       )
     }
   }
+
+  test("CANNOT_FIND_BATCH: cannot find batchMetadataFile") {
+    withTempPath { p =>
+      withSQLConf(
+        "spark.sql.streaming.checkpointFileManagerClass" ->
+          classOf[FileSystemBasedCheckpointFileManager].getName,
+        "fs.file.impl" -> classOf[FakeFileSystemNeverExists].getName,
+        "fs.file.impl.disable.cache" -> "true"
+      ) {
+        val checkpointLocation = p.getAbsolutePath
+        val batchMetadataFileName = new Path(checkpointLocation, "0")
+        val metadataLog = new HDFSMetadataLog[String](spark, checkpointLocation)
+        val e = intercept[SparkFileNotFoundException] (metadataLog.get(0))
```

Comment:
Could you try to trigger the error using public API. So, the idea is to write a batch, then read it back, and get the error because the fake filesystem returns an error from the `exists()` method.

Reply:
seems that the error is not easy to reproduce using public API, marking as internal error